doc/tutorials/troubleshooting.rst

```diff
@@ -0,0 +1,47 @@
+Troubleshooting
+===============
+
+Plugin errors
+-------------
+
+Errors that start with ``GHC Core to PLC plugin`` are ``plutus-tx-plugin`` errors.
+The most common error is:
+
+``Error: Reference to a name which is not a local, a builtin, or an external INLINABLE function``
+
+This means the plugin doesn't have access to implementation of the function, which it needs to be able to compile the function to Plutus Core.
+Some things you can do to fix it:
+
+- Make sure to add ``{-# INLINABLE functionname #-}`` to your function.
+- If there's an extra ``$c`` in front of the function name in the error, GHC has generated a specialised version of your function,
+  which prevents the plugin from accessing it.
+  You can turn it off with ``{-# OPTIONS_GHC -fno-specialise #-}``
+- Other compiler options that could fix the issue are
+
+  - ``{-# OPTIONS_GHC -fno-strictness #-}``
+  - ``{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}``
+  - ``{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}``
+  - ``{-# OPTIONS_GHC -fobject-code #-}``
+
+  Some more details are in `the plutus-tx readme <https://github.com/input-output-hk/plutus/tree/master/plutus-tx#building-projects-with-plutus-tx>`_.
```

Comment:
I think a great goal would be to get all the information in there into the docs!

Reply:
I thought this would be a waste of time since this won't be relevant anymore in the future?

Reply:
Well, one or the other :)